libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -0,0 +1,426 @@
+#include "vtr_assert.h"
+#include "vtr_memory.h"
+#include "vtr_util.h"
+
+#include "arch_types.h"
+#include "arch_util.h"
+#include "arch_error.h"
+
+#include "physical_types_util.h"
+
+struct t_pin_inst_port {
```

Comment:
Should comment what this data structure is for.  Gives data for a single physical pin I think -- what port it came from, what index within that port, what sub_tile and what capacity_instance?  Not quite sure what capacity_instance is -- which sub_tile within a certain type of sub_tile?

Reply:
Done